core/src/test/java/com/linecorp/armeria/server/cors/CorsServerErrorHanderTest.java

```diff
@@ -0,0 +1,89 @@
+
+public class CorsServerErrorHanderTest {
+    private static final ClientFactory clientFactory = ClientFactory.ofDefault();
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final HttpService myService = new AbstractHttpService() {
+                @Override
+                protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) {
+                    return HttpResponse.of(HttpStatus.OK);
+                }
+
+                @Override
+                protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req) {
+                    return HttpResponse.of(HttpStatus.OK);
+                }
+
+                @Override
+                protected HttpResponse doHead(ServiceRequestContext ctx, HttpRequest req) {
+                    return HttpResponse.of(HttpStatus.OK);
+                }
+
+                @Override
+                protected HttpResponse doOptions(ServiceRequestContext ctx, HttpRequest req) {
+                    return HttpResponse.of(HttpStatus.OK);
+                }
+            };
+            addCorsServiceWithException(sb, myService, "/cors_status_exception", HttpStatusException.of(HttpStatus.INTERNAL_SERVER_ERROR));
+            addCorsServiceWithException(sb, myService, "/cors_response_exception", HttpResponseException.of(HttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR)));
+        }
+    };
+
+    private static void addCorsServiceWithException(ServerBuilder sb, HttpService myService, String pathPattern, Exception exception) {
+        sb.service(pathPattern, myService.decorate(CorsService.builder("http://example.com").allowRequestMethods(HttpMethod.POST, HttpMethod.GET).allowRequestHeaders("allow_request_header").exposeHeaders("expose_header_1", "expose_header_2").preflightResponseHeader("x-preflight-cors", "Hello CORS").newDecorator()).decorate(new DecoratingHttpServiceFunction() {
+            @Override
+            public HttpResponse serve(HttpService delegate, ServiceRequestContext ctx, HttpRequest req) throws Exception {
+                throw exception;
+            }
+        }));
+    }
+
+    static WebClient client() {
+        return WebClient.builder(server.httpUri()).factory(clientFactory).build();
+    }
+
+    static AggregatedHttpResponse request(WebClient client, HttpMethod method, String path, String origin, String requestMethod) {
+        return client.execute(RequestHeaders.of(method, path, HttpHeaderNames.ACCEPT, "utf-8", HttpHeaderNames.ORIGIN, origin, HttpHeaderNames.ACCESS_CONTROL_REQUEST_METHOD, requestMethod)).aggregate().join();
+    }
+
+    static AggregatedHttpResponse preflightRequest(WebClient client, String path, String origin, String requestMethod) {
+        return request(client, HttpMethod.OPTIONS, path, origin, requestMethod);
+    }
+
+    void testCorsHeaderWithException(String path) {
```

Comment:
You could use a parameterized test instead?

```
@ParameterizedTest
@CsvSource({ "/cors_status_exception", "/cors_..." })
void testCorsHeadersWithException(String path) { ... }
```

Then you don't need to define two test methods.